topics/transcriptomics/tutorials/scrna_preprocessing/tutorial.md

```diff
@@ -0,0 +1,1385 @@
+
+> ### {% icon hands_on %} Hands-on: Confirming the Alignment Data
+>
+>  1. Click on the eye symbol of the BAM output from STAR.
+>  2. There are many header lines that begin with `@` which we are not interested in. 
+>  3. Do a Ctrl+F search for `@co` and then look at the lines directly below it.  
+>     One such read is given as so:
+> 
+>         J00182:75:HTKJNBBXX:2:1121:9729:45889_GACGAA_GTGGTC	16	chr1	2030	3	70M	*	0	0	AGAGGTTCCAATATTCCCATGAAATTGAGATTTTGTAAAAGAGTGAAGTGTGGTTACTTTCACTGAGAGG	JJJJJJJJJJJJJJJJJJJJJJJJFJJJJJAJJJJJJJJJJFJFJFFJJJJJJJJJJJJFF7AJA-77<A	NH:i:2 HI:i:1 AS:i:64 nM:i:2
+>
+{: .hands_on}
+
+The fields of the BAM file can be better explained at section 1.4 of [the SAM specification](https://samtools.github.io/hts-specs/SAMv1.pdf), but we will summarize the main fields of interest here:
+
+ * `J00182..._GACGAA_GCGGTC`: The *readname* appended by `_`, the cell barcode, another `_`, and then the UMI barcode.
+ * `16`: The FLAG value
+
+> ### {% icon question %} What does the FLAG value of 16 tell us about this read?
+>
+>   We can interactively see what the different FLAG values mean by feeding values into the SAM specification to the [picard web tool](https://broadinstitute.github.io/picard/explain-flags.html)
+>   > ### {% icon solution %} Solution
+>   > The read aligns to the reverse strand
+>   >
+>   {: .solution}
+>
+{: .question}
+
+ * `chr1` `2030`: The position and base-pair of alignment of the first base of the sequence.
+ * A series of quality fields, with the main contributors being  the sequence and sequence quality strings.
+ * `NH`: The number of hits for  this read. If it is multiply mapped, then the number of multiples will be shown (here `2`).
+ * `HI`: Which number this particular read is in the series of (potentially) multi-mapped reads (here `1`, not neccesarily meaning the first or 'better').
+ * `nM`: The number of mismatches in the alignment of this read to the reference (here `2`).
+
+
+#### Filtering the BAM File
+
+If we perform counting on the current BAM file we will be counting all reads, even the undesireable ones such as those that did not align so optimally.
+
+The main filtering steps performed on our reads so far have been relatively silent due to the 'default' parameters used.
+ * *UMI-tools Extract* - Filters reads for those only with matching barcodes given by our barcodes file.
+ * *RNA-STAR* - As seen in the log, we lose 10% of our reads for being too short or being multiply mapped.
+
+Another filtering measure we can apply is to keep reads that we are confident about, e.g those with a minimum number of mismatches to the reference within an acceptable range. 
+
+> ### {% icon hands_on %} Hands-on: Task description
+>
+> 1. **Filter** {% icon tool %} with the following parameters:
```

Comment:
You need to provide the full tool name here, or it will remain unclear, which you are refering to, so:

```suggestion
> 1. **Filter BAM datasets on a variety of attributes** {% icon tool %} with the following parameters:
```